BoxContentSDK/BoxContentSDK/External/HashHelper/BOXStreamingHashHelper.m

```diff
@@ -0,0 +1,48 @@
+//
+//  BOXStreamingHashHelper.m
+//  BoxContentSDK
+//
+//  Created by Prithvi Jutur on 5/21/18.
+//  Copyright © 2018 Box. All rights reserved.
+//
+
+#import "BOXHashHelper.h"
+
+#include <CommonCrypto/CommonDigest.h>
+#include <CoreFoundation/CoreFoundation.h>
+#import "BOXStreamingHashHelper.h"
+
+@interface BOXStreamingHashHelper()
+@property (nonatomic, readwrite, strong) NSString *calcSha1;
+@property (nonatomic, assign) CC_SHA1_CTX *context;
+@end
+
+@implementation BOXStreamingHashHelper
+- (instancetype) init {
+    _context = (CC_SHA1_CTX*) malloc(sizeof(CC_SHA1_CTX));
+    return self;
+}
+
+- (void) open {
+    CC_SHA1_Init(_context);
+}
+
+- (void) processData:(NSData* ) buffer{
+    CC_SHA1_Update(_context, [buffer bytes], (CC_LONG)buffer.length);
+}
+
+- (NSString *) close {
+    unsigned char digest[CC_SHA1_DIGEST_LENGTH];
+    CC_SHA1_Final(&digest[0], _context);
+
+    char hash[2 * sizeof(digest) + 1];
```

Comment:
I think it might be necessary to initialize this with `= ""` or `= {0}` in order to guarantee that it's zero-filled (the only exception, I believe, is for `static` variables). It's likely to be zeros by accident, so this would only show as a bug rarely, but could cause a EXC_BAD_ACCESS on the NSString initialization below.